x/trigger/keeper/event_listener_test.go

```diff
@@ -0,0 +1,158 @@
+package keeper_test
+
+import (
+	codectypes "github.com/cosmos/cosmos-sdk/codec/types"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdktx "github.com/cosmos/cosmos-sdk/types/tx"
+	"github.com/provenance-io/provenance/x/trigger/types"
+)
+
+func (s *KeeperTestSuite) TestGetAndSetEventListener() {
+	var event types.TriggerEventI = &types.BlockHeightEvent{BlockHeight: 1}
+	actions, _ := sdktx.SetMsgs([]sdk.Msg{&types.MsgDestroyTriggerRequest{Id: 1, Authority: s.accountAddresses[0].String()}})
+	eventAny, _ := codectypes.NewAnyWithValue(event)
+	newTrigger := types.NewTrigger(1, s.accountAddresses[0].String(), eventAny, actions)
+	s.app.TriggerKeeper.SetEventListener(s.ctx, newTrigger)
+
+	tests := []struct {
+		name     string
+		id       types.TriggerID
+		expected *types.Trigger
+		prefix   string
+		err      string
+	}{
+		{
+			name:     "valid - event listener",
+			id:       1,
+			expected: &newTrigger,
+			prefix:   types.BlockHeightPrefix,
+			err:      "",
+		},
+		{
+			name:     "invalid - event listener doesn't exist",
+			id:       999,
+			expected: nil,
+			prefix:   types.BlockHeightPrefix,
+			err:      types.ErrEventNotFound.Error(),
+		},
+		{
+			name:     "invalid - event listener doesn't exist with wrong prefix",
+			id:       1,
+			expected: nil,
+			prefix:   types.BlockTimePrefix,
+			err:      types.ErrEventNotFound.Error(),
+		},
+	}
+
+	for _, tc := range tests {
+		s.Run(tc.name, func() {
+			listener, err := s.app.TriggerKeeper.GetEventListener(s.ctx, tc.prefix, tc.id)
+			if len(tc.err) > 0 {
+				s.Error(err)
```

Comment:
Just a note. You should almost never be using `.Error(...)`. When an error is expected, you should almost always know at least some of what should be in the error. So `.EqualError` and `ErrorContains` should be used instead.

And if you do want to use `.Error(...)`, you MUST include `msgAndArgs` info. The failure message from `.Error(...)` is simply, "An error is expected but got nil." When the next person makes a change and gets that failure, they then have to read and understand what the test is doing in order to begin trying to figure out how their change broke it. But, for example, if you put the function name/call in the `msgAndArgs`, you'd immediately know the function call that didn't return an error.